### PR TITLE
test has not always led to the same result.

### PR DESCRIPTION
**Pull request by P8hJ**
This PR has two commits. 

1. when executing the actions, I realised that there is a test that is sometimes successful and sometimes not, depending on the test execution. I have refactored the test and production code so that the test now works reliably.

2. during the reorganisation I came across a method that took parameters that were not needed at all, I have now removed these.

### CODE REVIEW

File: src/test/java/org/eblocker/crypto/pki/PKITest.java [hunk 1]
```diff
@@ -160,18 +160,17 @@ public void testGenerateDeviceCertificate() throws Exception {
...
         int validityYears = 1;
         CertificateAndKey deviceRequest = PKI.generateSelfSignedCertificateRequest(name, KEYSIZE);
-        Date notValidAfter = DateUtil.addYears(new Date(), validityYears);
+        Date startDate = PKI.getStartDate();
+        Date notValidAfter = DateUtil.addYears(startDate, validityYears);
 
-        Date t1 = new Date();
-        X509Certificate deviceCertificate = PKI.generateSignedCertificate(deviceRequest.getCertificate(), ORGNAME, name, notValidAfter, l1ca);
-        Date t2 = new Date();
+        X509Certificate deviceCertificate = PKI.generateSignedCertificate(deviceRequest.getCertificate(), startDate, notValidAfter, l1ca);
```
Comment (P8hJ) [inline src/test/java/org/eblocker/crypto/pki/PKITest.java:166]: This is the problematic test that worked once and did not work once.

File: src/main/java/org/eblocker/crypto/pki/PKI.java [hunk 2]
```diff
@@ -244,23 +244,22 @@ public static CertificateAndKey generateSelfSignedCertificateRequest(String name
...
+        return doGenerateSignedCertificate(request, getStartDate(), notValidAfter, l1ca, KeyPurposeId.id_kp_clientAuth);
     }
 
-    public static X509Certificate generateSignedCertificate(X509Certificate request, String orgName, String name, Date notValidAfter, CertificateAndKey l1ca) throws CryptoException {
-        return doGenerateSignedCertificate(request, orgName, name, notValidAfter, l1ca);
+    public static X509Certificate generateSignedCertificate(X509Certificate request, Date startDate, Date notValidAfter, CertificateAndKey l1ca) throws CryptoException {
+        return doGenerateSignedCertificate(request, startDate, notValidAfter, l1ca);
     }
 
-    private static X509Certificate doGenerateSignedCertificate(X509Certificate request, String orgName, String name, Date notValidAfter, CertificateAndKey l1ca, KeyPurposeId...keyPurposeIds) throws CryptoException {
```
Comment (P8hJ) [inline src/main/java/org/eblocker/crypto/pki/PKI.java:259]: These two parameters orgName and name are not used in this method. I have therefore deleted them.

Comment (eblosug) [reply src/main/java/org/eblocker/crypto/pki/PKI.java:259]: OK, I think it's a good idea to get rid of unused parameters.

The only thing I would suggest is to use the same parameters in these very similar methods:

* `generateTLSServerCertificate()`
* `generateTLSClientCertificate()`
* `generateSignedCertificate()`

So I would add the `startDate` parameter to `generateTLSServerCertificate()` and `generateTLSClientCertificate()`, too.

What do you think?

Comment (P8hJ) [reply src/main/java/org/eblocker/crypto/pki/PKI.java:259]: Many thanks for the tip. I'm always happy to receive tips like that. I have added another commit. In this commit I have also changed the call of static methods via the class variable and not via the instance variable.

**Pull request merged**